### PR TITLE
Fix "Session object destruction failed" - update

### PR DESCRIPTION
**Pull request by luhla**

### CODE REVIEW

**Comment by luhla** [comment]
Sorry, have to study how run tests before pull request...

**Comment by hrach** [comment]
And try to fix the current one, not to open another :)

File: src/Http/Session.php [hunk 1]
```diff
@@ -217,9 +217,9 @@ public function regenerateId()
 				throw new Nette\InvalidStateException('Cannot regenerate session ID after HTTP headers have been sent' . ($file ? " (output started at $file:$line)." : '.'));
 			}
 			if (session_id() !== '') {
+				session_write_close();
```
Comment (enumag) [inline src/Http/Session.php]: Maybe the `session_write_close();` should be before the if. That should fix the failing test as well.

Comment (luhla) [inline src/Http/Session.php]: I did try it before, but result was also failed checks. Solution is session_write_close(); before session_regenerate_id(TRUE); in my case but I wouldnt do annoing tests here.

**Comment by dg** [comment]
Can you explain what is it good for?

**Comment by luhla** [comment]
Getting "session_regenerate_id(): Session object destruction failed" when calling before session_write_close(); 
But i can`t call session_write_close(); before session_id().

**Comment by dg** [comment]
Is it needed to call session_write_close() when session_id() === '' ?

**Comment by luhla** [comment]
Getting error when using:
```
if (session_id() !== '') {
        session_regenerate_id(TRUE);
}
```
Everithing fine when using: 
```
if (session_status() === PHP_SESSION_ACTIVE ) {
        session_regenerate_id(TRUE);
}
```

**Comment by dg** [comment]
IMHO session_write_close(); should be in condition too.

**Comment by dg** [comment]
Did you change the order of session_write_close and session_regenerate_id intentionally?

**Comment by luhla** [comment]
Make no sense for me, where should be session_write_close()  "in condition too" ? 
After session_regenerate_id(TRUE) it is unnecessary because there is another session_write_close() after condition. But it should be always before $backup = $_SESSION; 
Giving session_write_close() before session_regenerate_id(TRUE) is not obviously good idea.

**Comment by dg** [comment]
> After session_regenerate_id(TRUE) it is unnecessary because there is another session_write_close() after condition

Why must be another after condition?

**Comment by luhla** [comment]
Well, thats true, it might not, when seesion status is not active, nothing will be writen to session. I wil fix it.

**Comment by dg** [comment]
Thanks!

**Comment by luhla** [comment]
Thanks for nothing, you did the solution. :-)

**Pull request merged**